### PR TITLE
Missing fixes

### PR DESCRIPTION
**Pull request by victornoel**
A bunch of fixes related to previous issues, see commits for details.

### CODE REVIEW

File: src/main/java/org/cactoos/scalar/AvgOf.java [hunk 1]
```diff
@@ -136,22 +129,17 @@ public AvgOf(final Scalar<Number>... src) {
     public AvgOf(final Iterable<Scalar<Number>> src) {
         super(
             new Ternary<>(
-                new LengthOf(src),
-                (Double len) -> len > 0,
-                len -> new Folded<BigDecimal, BigDecimal>(
+                new ScalarOf<>(() -> new ListOf<>(new Mapped<Number>(Scalar::value, src))),
+                (List<Number> list) -> list.size() > 0,
```
Comment (andreoss) [inline src/main/java/org/cactoos/scalar/AvgOf.java:133]: @victornoel `!list.isEmpty()`

**Review by andreoss — CHANGES_REQUESTED** [review]

**Comment by victornoel** [comment]
@andreoss thank you, I applied the suggestion :)

**Comment by victornoel** [comment]
@andreoss is this ok for you now?

**Review by andreoss — APPROVED** [review]

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @andreoss[/z](https://www.0crat.com/u/andreoss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@andreoss please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable  ([here](https://github.com/yegor256/cactoos/issues/1516#issuecomment-752751341))

@sereshqua There is no quality review for [#1516](https://github.com/yegor256/cactoos/issues/1516), no performer

**Pull request merged**